Suppression of warnings about possibly uninitialized values is required.

The amount of warnings issued during the compilation is not very large, but each warning is long and the output is completely unreadable
```
[ 18%] Building CXX object Tests/UnitTests/Core/Definitions/CMakeFiles/ActsUnitTestUnits.dir/UnitsTests.cpp.o
In file included from /Projects/acts/Tests/IntegrationTests/Autodiff/PropagationCompareAtlasAutodiffConstant.cpp:15:
In member function ‘bool Acts::AutodiffExtensionWrapper<basic_extension_t>::k(const propagator_state_t&, const stepper_t&, Acts::Vector3&, const Vector3&, std::array<double, 4>&, int, double, const Vector3&) [with propagator_state_t = Acts::Propagator<Acts::EigenStepper<Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> > > >::State<Acts::PropagatorOptions<Acts::ActionList<>, Acts::AbortList<Acts::PathLimitReached> > >; stepper_t = Acts::EigenStepper<Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> > >; basic_extension_t = Acts::detail::GenericDefaultExtension]’,
    inlined from ‘Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> >::k<Acts::Propagator<Acts::EigenStepper<Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> > > >::State<Acts::PropagatorOptions<Acts::ActionList<>, Acts::AbortList<Acts::PathLimitReached> > >, Acts::EigenStepper<Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> > > >(const Acts::Propagator<Acts::EigenStepper<Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> > > >::State<Acts::PropagatorOptions<Acts::ActionList<>, Acts::AbortList<Acts::PathLimitReached> > >&, const Acts::EigenStepper<Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> > >&, Acts::Vector3&, const Vector3&, std::array<double, 4>&, int, double, const Vector3&)::<lambda(auto:41, auto:42&)> [with auto:41 = std::integral_constant<int, 1>; auto:42 = Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> >::k<Acts::Propagator<Acts::EigenStepper<Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> > > >::State<Acts::PropagatorOptions<Acts::ActionList<>, Acts::AbortList<Acts::PathLimitReached> > >, Acts::EigenStepper<Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> > > >(const Acts::Propagator<Acts::EigenStepper<Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> > > >::State<Acts::PropagatorOptions<Acts::ActionList<>, Acts::AbortList<Acts::PathLimitReached> > >&, const Acts::EigenStepper<Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> > >&, Acts::Vector3&, const Vector3&, std::array<double, 4>&, int, double, const Vector3&)::<lambda(auto:41, auto:42&)>]’ at /Projects/acts/Core/include/Acts/Propagator/StepperExtensionList.hpp:96:28,
    inlined from ‘bool Acts::StepperExtensionList<extensions>::k(const propagator_state_t&, const stepper_t&, Acts::Vector3&, const Vector3&, std::array<double, 4>&, int, double, const Vector3&) [with propagator_state_t = Acts::Propagator<Acts::EigenStepper<Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> > > >::State<Acts::PropagatorOptions<Acts::ActionList<>, Acts::AbortList<Acts::PathLimitReached> > >; stepper_t = Acts::EigenStepper<Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> > >; extensions = {Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension>}]’ at /Projects/acts/Core/include/Acts/Propagator/StepperExtensionList.hpp:105:16,
    inlined from ‘bool Acts::StepperExtensionList<extensions>::k1(const propagator_state_t&, const stepper_t&, Acts::Vector3&, const Vector3&, std::array<double, 4>&) [with propagator_state_t = Acts::Propagator<Acts::EigenStepper<Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> > > >::State<Acts::PropagatorOptions<Acts::ActionList<>, Acts::AbortList<Acts::PathLimitReached> > >; stepper_t = Acts::EigenStepper<Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> > >; extensions = {Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension>}]’ at /Projects/acts/Core/include/Acts/Propagator/StepperExtensionList.hpp:115:13,
    inlined from ‘Acts::Result<double> Acts::EigenStepper<extensionlist_t, auctioneer_t>::step(propagator_state_t&) const [with propagator_state_t = Acts::Propagator<Acts::EigenStepper<Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> > > >::State<Acts::PropagatorOptions<Acts::ActionList<>, Acts::AbortList<Acts::PathLimitReached> > >; extensionlist_t = Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> >; auctioneer_t = Acts::detail::VoidAuctioneer]’ at /Projects/acts/Core/include/Acts/Propagator/EigenStepper.ipp:126:35:
/Projects/acts/Plugins/Autodiff/include/Acts/Plugins/Autodiff/AutodiffExtensionWrapper.hpp:53:31: warning: ‘<anonymous>’ may be used uninitialized [-Wmaybe-uninitialized]
   53 |     return m_doubleExtension.k(state, stepper, knew, bField, kQoP, i, h, kprev);
      |            ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /Projects/acts/Core/include/Acts/Propagator/DefaultExtension.hpp:11,
                 from /Projects/acts/Tests/IntegrationTests/Autodiff/PropagationCompareAtlasAutodiffConstant.cpp:17:
/Projects/acts/Core/include/Acts/Propagator/detail/GenericDefaultExtension.hpp: In member function ‘Acts::Result<double> Acts::EigenStepper<extensionlist_t, auctioneer_t>::step(propagator_state_t&) const [with propagator_state_t = Acts::Propagator<Acts::EigenStepper<Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> > > >::State<Acts::PropagatorOptions<Acts::ActionList<>, Acts::AbortList<Acts::PathLimitReached> > >; extensionlist_t = Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> >; auctioneer_t = Acts::detail::VoidAuctioneer]’:
/Projects/acts/Core/include/Acts/Propagator/detail/GenericDefaultExtension.hpp:54:8: note: by argument 7 of type ‘const ThisVector3&’ {aka ‘const Eigen::Matrix<double, 3, 1>&’} to ‘bool Acts::detail::GenericDefaultExtension<scalar_t>::k(const propagator_state_t&, const stepper_t&, Acts::detail::GenericDefaultExtension<scalar_t>::ThisVector3&, const Vector3&, std::array<scalar_t, 4>&, int, double, const ThisVector3&) [with propagator_state_t = Acts::Propagator<Acts::EigenStepper<Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> > > >::State<Acts::PropagatorOptions<Acts::ActionList<>, Acts::AbortList<Acts::PathLimitReached> > >; stepper_t = Acts::EigenStepper<Acts::StepperExtensionList<Acts::AutodiffExtensionWrapper<Acts::detail::GenericDefaultExtension> > >; scalar_t = double]’ declared here
   54 |   bool k(const propagator_state_t& state, const stepper_t& stepper,
      |        ^
In file included from /Projects/acts/Core/include/Acts/Propagator/EigenStepper.hpp:19,
                 from /Projects/acts/Tests/IntegrationTests/Autodiff/PropagationCompareAtlasAutodiffConstant.cpp:18:
/Projects/acts/Core/include/Acts/Propagator/StepperExtensionList.hpp:115:13: note: ‘<anonymous>’ declared here
  115 |     return k(state, stepper, knew, bField, kQoP, 0);
```
CXX:  gcc version 11.1.1 20210531 (Red Hat 11.1.1-3) (GCC)
with standard flags.


Best regards,

Andrii

--- a/Core/include/Acts/Propagator/StepperExtensionList.hpp
+++ b/Core/include/Acts/Propagator/StepperExtensionList.hpp
@@ -79,7 +79,8 @@ struct StepperExtensionList : private detail::Extendable<extensions...> {
   template <typename propagator_state_t, typename stepper_t>
   bool k(const propagator_state_t& state, const stepper_t& stepper,
          Vector3& knew, const Vector3& bField, std::array<double, 4>& kQoP,
-         const int i, const double h = 0., const Vector3& kprev = Vector3{}) {
+         const int i, const double h = 0.,
+         const Vector3& kprev = Vector3::Zero()) {
     // TODO replace with integer-templated lambda with C++20
     auto impl = [&, i, h](auto intType, auto& implRef) {
       constexpr int N = decltype(intType)::value;

--- a/Core/include/Acts/Propagator/detail/GenericDefaultExtension.hpp
+++ b/Core/include/Acts/Propagator/detail/GenericDefaultExtension.hpp
@@ -54,7 +54,7 @@ struct GenericDefaultExtension {
   bool k(const propagator_state_t& state, const stepper_t& stepper,
          ThisVector3& knew, const Vector3& bField, std::array<Scalar, 4>& kQoP,
          const int i = 0, const double h = 0.,
-         const ThisVector3& kprev = ThisVector3()) {
+         const ThisVector3& kprev = ThisVector3::Zero()) {
     auto qop =
         stepper.charge(state.stepping) / stepper.momentum(state.stepping);
     // First step does not rely on previous data

--- a/Core/include/Acts/Propagator/detail/GenericDenseEnvironmentExtension.hpp
+++ b/Core/include/Acts/Propagator/detail/GenericDenseEnvironmentExtension.hpp
@@ -101,7 +101,7 @@ struct GenericDenseEnvironmentExtension {
   bool k(const propagator_state_t& state, const stepper_t& stepper,
          ThisVector3& knew, const Vector3& bField, std::array<Scalar, 4>& kQoP,
          const int i = 0, const double h = 0.,
-         const ThisVector3& kprev = ThisVector3()) {
+         const ThisVector3& kprev = ThisVector3::Zero()) {
     // i = 0 is used for setup and evaluation of k
     if (i == 0) {
       // Set up container for energy loss

--- a/Plugins/Autodiff/include/Acts/Plugins/Autodiff/AutodiffExtensionWrapper.hpp
+++ b/Plugins/Autodiff/include/Acts/Plugins/Autodiff/AutodiffExtensionWrapper.hpp
@@ -49,7 +49,7 @@ struct AutodiffExtensionWrapper {
   bool k(const propagator_state_t& state, const stepper_t& stepper,
          Vector3& knew, const Vector3& bField, std::array<double, 4>& kQoP,
          const int i = 0, const double h = 0.,
-         const Vector3& kprev = Vector3()) {
+         const Vector3& kprev = Vector3::Zero()) {
     return m_doubleExtension.k(state, stepper, knew, bField, kQoP, i, h, kprev);
   }
 

